ocaml-eglot-type-enclosing.el

```diff
@@ -0,0 +1,171 @@
+    (define-key keymap (kbd "C-<right>") #'ocaml-eglot-type-enclosing-increase-verbosity)
+    (define-key keymap (kbd "C-<left>") #'ocaml-eglot-type-enclosing-decrease-verbosity)
+    keymap)
+  "Keymap for OCaml-eglot's type enclosing transient mode.")
+
+;;; Internal functions
+
+(defun ocaml-eglot-type-enclosing-copy ()
+  "Copy the type of the current enclosing to the Kill-ring."
+  (interactive)
+  (when ocaml-eglot-type-enclosing-current-type
+    (eglot--message "Copied `%s' to kill-ring"
+                    ocaml-eglot-type-enclosing-current-type)
+    (kill-new ocaml-eglot-type-enclosing-current-type)))
+
+(defun ocaml-eglot-type-enclosing--with-fixed-offset ()
+  "Compute the type enclosing for a dedicated offset."
+  (let* ((verbosity ocaml-eglot-type-enclosing-verbosity)
+         (index ocaml-eglot-type-enclosing-offset)
+         (at (ocaml-eglot-util--current-position-or-range))
+         (result (ocaml-eglot-req--type-enclosings at index verbosity))
+         (type (cl-getf result :type)))
+    (setq ocaml-eglot-type-enclosing-current-type type)
+    (ocaml-eglot-type-enclosing--display type t)))
+
+(defun ocaml-eglot-type-enclosing-increase-verbosity ()
+  "Increase the verbosity of the current request."
+  (interactive)
+  (if ocaml-eglot-type-enclosing-verbosity
+      (setq ocaml-eglot-type-enclosing-verbosity
+            (1+ ocaml-eglot-type-enclosing-verbosity))
+    (setq ocaml-eglot-type-enclosing-verbosity 1))
+  (ocaml-eglot-type-enclosing--with-fixed-offset))
+
+(defun ocaml-eglot-type-enclosing-decrease-verbosity ()
+  "Decrease the verbosity of the current request."
+  (interactive)
+  (if ocaml-eglot-type-enclosing-verbosity
+      (progn
+        (setq ocaml-eglot-type-enclosing-verbosity
+              (1- ocaml-eglot-type-enclosing-verbosity))
+        (when (< ocaml-eglot-type-enclosing-verbosity 1)
+          (setq ocaml-eglot-type-enclosing-verbosity nil)))
+    (setq ocaml-eglot-type-enclosing-verbosity nil))
+  (ocaml-eglot-type-enclosing--with-fixed-offset))
+
+(defun ocaml-eglot-type-enclosing-grow ()
+  "Growing of the type enclosing."
+  (interactive)
+  (when ocaml-eglot-type-enclosing-types
+    (if (>= ocaml-eglot-type-enclosing-offset
+            (1- (length ocaml-eglot-type-enclosing-types)))
+        (setq ocaml-eglot-type-enclosing-offset 0)
+      (setq ocaml-eglot-type-enclosing-offset
+            (1+ ocaml-eglot-type-enclosing-offset)))
+    (ocaml-eglot-type-enclosing--with-fixed-offset)))
+
+(defun ocaml-eglot-type-enclosing-shrink ()
+  "Display the type enclosing of a smaller enclosing if possible."
+  (interactive)
+  (when ocaml-eglot-type-enclosing-types
+    (if (<= ocaml-eglot-type-enclosing-offset 0)
+        (setq ocaml-eglot-type-enclosing-offset
+              (1- (length ocaml-eglot-type-enclosing-types)))
+      (setq ocaml-eglot-type-enclosing-offset
+            (1- ocaml-eglot-type-enclosing-offset)))
```

Comment:
I find that indentation weird, but you do it for every if so I guess it is correct ? (I would expect the if branches to be at the same level)

Also, it's a a good idea to loop to the biggest enclosing, and that gives us an opportunity to "simplify" these functions by using a modulo, doesn't it ?

Reply:
> (I would expect the if branches to be at the same level)

It is the case but since `1-` is a function, the opening parens give a strange look. 

> Also, it's a a good idea to loop to the biggest enclosing, and that gives us an opportunity to "simplify" these functions by using a modulo, doesn't it ?

I'll try!

Reply:
I was really speaking of the two branches of the if:
```
    (if (<= ocaml-eglot-type-enclosing-offset 0)
        (setq ocaml-eglot-type-enclosing-offset
              (1- (length ocaml-eglot-type-enclosing-types)))
        (setq ocaml-eglot-type-enclosing-offset
              (1- ocaml-eglot-type-enclosing-offset)))
```

instead of 

```
    (if (<= ocaml-eglot-type-enclosing-offset 0)
        (setq ocaml-eglot-type-enclosing-offset
              (1- (length ocaml-eglot-type-enclosing-types)))
      (setq ocaml-eglot-type-enclosing-offset
            (1- ocaml-eglot-type-enclosing-offset)))```
```

(same for other if in the file)

Reply:
Ah yes, it is the formatting provided by `indent-region` :/ It is the convention : 

```scheme
(if pred
    if-true
  if-false)
```